app/auth/gitlab_auth.py

```diff
@@ -40,198 +36,80 @@
     url_for,
 )
 
-from .web import JWT_ALGORITHM, get_key_for_user
+from .oauth_provider_app import GitLabProviderApp
+from .utils import (
+    get_redis_key_from_session,
+    get_redis_key_from_token,
+    handle_login_request,
+    handle_token_request,
+)
+from .oauth_client import RenkuWebApplicationClient
+
+GL_SUFFIX = "gl_oauth_client"
 
 blueprint = Blueprint("gitlab_auth", __name__, url_prefix="/auth/gitlab")
 
 
+# Note: GitLab oauth tokens do NOT expire per default
+# See https://gitlab.com/gitlab-org/gitlab/-/issues/21745
+# The documentation about this is wrong
+# (https://docs.gitlab.com/ce/api/oauth2.html#web-application-flow)
+
+
 class GitlabUserToken:
     def __init__(self, header_field="Authorization", header_prefix="Bearer "):
         self.header_field = header_field
         self.header_prefix = header_prefix
 
     def process(self, request, headers):
```

Comment:
Could we add some docs here?

Reply:
Added an issue for refactoring that whole part: https://github.com/SwissDataScienceCenter/renku-gateway/issues/232